Update on python API to correct undefined references

jmllorens opened the pull request:
Since the migration to the new C API the python module stop working as expected. A few errors appear after import: 

- undefined functions: `Layer_Init`, `Material_Init`, `Material_InitTensor`, `Simulation_AddMaterial`, `Simulation_GetPoyntingFlux` and `Simulation_MakeExcitationExterior`
- The `SetFrequency` stop detroying previous results.

All these dependencies to the old API have been corrected by upgrading to the new C API when possible.

Old API calls still remains, although the python extension is more useful that the current version.

jmllorens committed:
Update on python API to correct undefined references

Since the migration to the new C API the python module stop working as expected. A few errors appear after import: undefined functions (Layer_Init, Material_Init, Material_InitTensor, Simulation_AddMaterial, Simulation_GetPoyntingFlux and Simulation_MakeExcitationExterior) and the SetFrequency stop detroying the previous results. All these dependencies to the old API have been corrected by upgrading to the new C API when possible.

Commit:
Merge branch 'python'

Commit:
Modified the python file

Patched S4Sim_SetRegionCircle file

kwrobert commented:
Works like a charm! Thanks so much for making these corrections.

jmllorens commented:
@kwrobert  After sending the pull request I noticed that some of functions which still use the old API are not working properly.  For example, I fixed `S4Sim_SetRegionCircle` in my fork, because is one of the functions I use most. Probably it is a better a idea to wait until `S4v2` is merged with `master` to implement all the changes.

jmllorens committed:
Merge remote-tracking branch 'upstream/master'

jmllorens committed:
Auto stash before merge of "python" and "master"

jmllorens committed:
Updated python API 

The update of the python API is complete for regions. Apparently, everything is working. The only difference between these results and those of ca2b90 commit are when the region has an angle different from zero.

Changes in pattern.c are to make it compliant with SuiteSparse. 

The make related files are forced to use the system gcc for compilation and linking.

jmllorens committed:
Downgrade of kwlist

The kwlist of the python API is compatible with the old version. S4_Layer and S4_Material are changed to Layer and Material.

jmllorens committed:
Merge remote-tracking branch 'upstream/master' into python

jmllorens committed:
Main_python.c compatible with python3

Chronum94 commented:
I get a 
```ImportError: /home/chronum/S4/jmllorens/S4/build/lib.linux-x86_64-3.6/S4.cpython-36m-x86_64-linux-gnu.so: undefined symbol: PyInt_AsLong``` when working with this. Any ideas?

jmllorens committed:
Trace option

Commented the ENABLE_S4_TRACE option, which is useful for debugging purposes.

jmllorens committed:
Maintenance 

Binaries removed from index and ENABLE_S4_TRACE removed as default option in Makefile.ubuntu.

jmllorens committed:
Merge remote-tracking branch 'origin/python' into python

jmllorens committed:
Modula name changed to S42

This branch uses the new S4 python API which turns out to be very unstable. To coexists both modules, the name of the module in this branch changed from S4 to S42.
